pkg/operator/operator.go

```diff
@@ -469,18 +469,30 @@ func (o *Operator) sync(key string) error {
 		klog.Errorf("error occurred while setting status to in progress: %v", err)
 	}
 
+	promOpTaskName := "Updating Prometheus Operator"
```

Comment:
what if `TaskRunner` can take an ordered list of tasks? e.g. 

```golang
tl := NewTaskRunner(client *client.Client)

# first group of tasks to execute.
tl.AddTasks(
    tasks.NewTaskSpec("Updating Prometheus Operator", tasks.NewPrometheusOperatorTask(o.client, factory)),
)
# second group of tasks to execute.
tl.AddTasks(
    tasks.NewTaskSpec("Updating user workload Prometheus Operator", tasks.NewPrometheusOperatorUserWorkloadTask(o.client, factory, config)),
    ...
)

taskName, err := tl.RunAll()
```

Reply:
Thought about it, but figured as long as we only have one task to run before, I might as well use whats already there.

Reply:
Thinking about it more, I'd rather see us only reuse TaskRunner (even for only one task):

```golang
# first group of tasks to execute
tl := NewTaskRunner(o.client,[]*tasks.TaskSpec{
    tasks.NewTaskSpec("Updating Prometheus Operator", tasks.NewPrometheusOperatorTask(o.client, factory)),
})

taskName, err := tl.RunAll()
...

# second group of tasks to execute.
tl = NewTaskRunner(o.client,[]*tasks.TaskSpec{
    tasks.NewTaskSpec("Updating user workload Prometheus Operator", tasks.NewPrometheusOperatorUserWorkloadTask(o.client, factory, config)),
    ...
)

taskName, err = tl.RunAll()
...
```

Reply:
Opted for the first idea of a list of task lists. This way we don't need to add error handling if the first task list has a failed task.